javaee/rt/src/java/foundation/icon/ee/util/bls12381/BLS12381.java

```diff
@@ -79,4 +86,122 @@ public static boolean verifyG2Signature(byte[] pubKey, byte[] sig, byte[] msg) {
             throw new IllegalArgumentException(e);
         }
     }
+
+    private static byte[] concat(byte[]... args) {
+        int length = 0;
+        for (int i = 0; i < args.length; i++) {
+            length += args[i].length;
+        }
+        byte[] out = new byte[length];
+        int offset = 0;
+        for (int i = 0; i < args.length; i++) {
+            System.arraycopy(args[i], 0, out, offset, args[i].length);
+            offset += args[i].length;
+        }
+        return out;
+    }
+
+    public static byte[] g1Add(byte[] data, boolean compressed) {
+        P1 acc = new P1();
+        int size = compressed ? G1_LEN : 2 * G1_LEN;
+        if (data.length == 0 || data.length % size != 0) {
+            throw new IllegalArgumentException("BLS12-381: g1Add: invalid data layout: expected a multiple of " + size
+                    + " bytes, got " + data.length);
+        }
+        byte[] buf = new byte[size];
+        for (int i = 0; i < data.length; i += size) {
+            System.arraycopy(data, i, buf, 0, size);
+            acc = acc.add(new P1(buf));
+        }
+        return compressed ? acc.compress() : acc.serialize();
+    }
+
+    public static byte[] flipUncompressedG2Layout(byte[] data) {
```

Comment:
Can you explain why this method is necessary for uncompressed format?

Reply:
The layout of a G2 point that BLST JNI takes is of the form, 

BLST: (x1 * i + x0, y1 * i + y0) -> (x1, x0, y1, y0)

The input to the the Context API, however, has to be made consistent and equivalent to the norm (similar to what ethereum pre-compiles expose), so that there is consistency between how inputs are provided for different curves. We've been testing multiple curves, but this PR is specific to BLS. I could add comment in the code specifying the reason.

The input format has been specified in the Context API's comments for each of the exposed methods.